rest_docs/rest-catalog-open-api.yaml

```diff
@@ -1381,15 +1381,35 @@ components:
               } }
 
     CreateNamespaceResponse:
-      description: A succesful call to create a namespace
+      description:
+        Represents a successful call to create a namespace.
```

Comment:
Since this is basically the same as `GetNamespaceResponse`, can we combine the two into `GetOrCreateNamespaceResponse` and use it in both places?

Reply:
So I'm of two opinions on this:
1) I can agree with the argument that I believe @danielcweeks made in [a related PR](https://github.com/apache/iceberg/pull/4037#discussion_r799095408) about allowing both types to evolve separately by keeping them separate.
2) However, I also think it wouldn't be too hard to update later on to split them out if we needed to for evolving them.

The big thing I'd give consideration to is whether or not additional metadata is being returned by these responses (that isn't defined in the spec). This would allow implementors to enrich the calls with their own details (for example, span IDs for distributed tracing end to end).

But since we don't support that currently, I would still prefer to keep them separate to evolve separately. But I don't have any issues with changing it to be `GetOrCreateNamespaceResponse`, as updating it to be two classes in the future (if ever needed) should only involve a handful of location updates and we then have less unit tests etc to write.

Any thoughts on this @danielcweeks?

Reply:
Sounds fine to keep them separate.